src/content/app/tools/blast/components/listed-blast-submission/ListedBlastSubmission.test.tsx

```diff
@@ -0,0 +1,235 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Provider } from 'react-redux';
+import { MemoryRouter } from 'react-router';
+import { configureStore, combineReducers } from '@reduxjs/toolkit';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import * as blastStorageService from 'src/content/app/tools/blast/services/blastStorageService';
+
+import ListedBlastSubmission, {
+  type Props
+} from 'src/content/app/tools/blast/components/listed-blast-submission/ListedBlastSubmission';
+
+import blastFormReducer from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+import blastResultsReducer, {
+  type BlastResultsState
+} from 'src/content/app/tools/blast/state/blast-results/blastResultsSlice';
+
+import { createBlastSubmission } from 'tests/fixtures/blast/blastSubmission';
+
+jest.mock('src/content/app/tools/blast/services/blastStorageService');
+
+const defaultProps = {
+  submission: createBlastSubmission()
+};
+
+const renderComponent = ({
+  props,
+  state
+}: {
+  props?: Partial<Props>;
+  state?: Partial<BlastResultsState>;
+}) => {
+  const initialState = {
+    blast: { blastResults: state ?? {} }
+  };
+
+  const mergedProps = {
+    ...defaultProps,
+    ...props
+  };
+  const rootReducer = combineReducers({
+    blast: combineReducers({
+      blastForm: blastFormReducer,
+      blastResults: blastResultsReducer
+    })
+  });
+
+  const store = configureStore({
+    reducer: rootReducer,
+    preloadedState: initialState
+  });
+
+  const renderResult = render(
+    <MemoryRouter>
+      <Provider store={store}>
+        <ListedBlastSubmission {...mergedProps} />
+      </Provider>
+    </MemoryRouter>
+  );
+
+  return {
+    ...renderResult,
+    store
+  };
+};
+
+describe('BlastSubmissionHeader', () => {
+  describe('rendering', () => {
+    it('shows one sequence box if the submission contained a single sequence', () => {
+      const submission = createBlastSubmission({
+        options: { sequencesCount: 1 }
+      });
+
+      const { container } = renderComponent({
+        props: { submission }
+      });
+
+      expect(container.querySelectorAll('.sequenceBox').length).toBe(1);
+    });
+
+    it('shows multiple sequence boxes if the submission contained multiple sequences', () => {
+      const submission = createBlastSubmission({
+        options: { sequencesCount: 5 }
+      });
+
+      const { container } = renderComponent({
+        props: { submission }
+      });
+
+      expect(container.querySelectorAll('.sequenceBox').length).toBe(5);
+    });
+
+    it.todo('shows BLAST program');
+
+    it.todo('shows submission id');
+
+    it.todo('shows submission date');
```

Comment:
Is this for a separate PR?

Reply:
I was wondering whether to include this in the tests at all. Seems pretty minor. What do you think?

Reply:
I think it would still be worth writing a simple test case for the `getDateString` method. I think it can be moved to a shared helper file(as you have mentioned in a TODO)  and tested there instead?

Reply:
Yep, makes sense. I moved the code for formatting the date into a separate module, and added tests for the formatting. Removed these three todos from `ListBlastSubmission.test.ts`.